In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("Mental health in tech survey 2014-2019.csv")
df

year self_employed number_of_employees tech_company  \
0     2014            No             100-500          Yes   
1     2014            No              26-100          Yes   
2     2014            No      More than 1000           No   
3     2014            No              26-100          Yes   
4     2014            No      More than 1000           No   
...    ...           ...                 ...          ...   
3478  2019            No              26-100          Yes   
3479  2019            No                6-25          Yes   
3480  2019            No      More than 1000           No   
3481  2019            No      More than 1000          Yes   
3482  2019            No      More than 1000          Yes   

     mental_health_benefits mental_health_benefits_awareness  \
0                       Yes                              Yes   
1                       Yes                              Yes   
2                Don't know                               No   
3                       Yes                         Not sure   
4                       Yes                              Yes   
...                     ...                              ...   
3478                    Yes                               No   
3479                     No                               No   
3480             Don't know                               No   
3481                     No                               No   
3482                    Yes                              Yes   

     employer_mental_health_discussion  \
0                                   No   
1                                   No   
2                                   No   
3                           Don't know   
4                                   No   
...                                ...   
3478                                No   
3479                                No   
3480                                No   
3481                                No   
3482                                No   

     employer_mental_health_learning_resources  \
0                                           No   
1                                           No   
2                                   Don't know   
3                                          Yes   
4                                   Don't know   
...                                        ...   
3478                                        No   
3479                                        No   
3480                                Don't know   
3481                                        No   
3482                                       Yes   

     mental_health_treatment_anonymity mental_health_leave_accessibility  \
0                                   No                Somewhat difficult   
1                           Don't know                        Don't know   
2                           Don't know                        Don't know   
3                                  Yes                        Don't know   
4                                   No                         Very easy   
...                                ...                               ...   
3478                               Yes                         Very easy   
3479                        Don't know                         Very easy   
3480                        Don't know                Somewhat difficult   
3481                               Yes                         Difficult   
3482                               Yes                Somewhat difficult   

     mental_health_discussion_comfort_coworkers  \
0                                  Some of them   
1                                  Some of them   
2                                            No   
3                                           Yes   
4                                  Some of them   
...                                         ...   
3478                                        Yes   
3479                                         No   
3480                      

In [3]:
df_2014 = df[df["year"]==2014]
shape14 = df_2014.shape[0]
df_2016 = df[df["year"]==2016]
shape16 = df_2016.shape[0]
df_2017 = df[df["year"]==2017]
shape17 = df_2017.shape[0]
df_2018 = df[df["year"]==2018]
shape18 = df_2018.shape[0]
df_2019 = df[df["year"]==2019]
shape19 = df_2019.shape[0]

In [4]:
# whether their employer provides mental health benefits as part of healthcare coverage
mentalhealthcarecoverage_2014 = df_2014["mental_health_benefits"].value_counts()/shape14
mentalhealthcarecoverage_2014

mental_health_benefits
Yes           0.414161
Don't know    0.339476
No            0.246363
Name: count, dtype: float64

In [5]:
mentalhealthcarecoverage_2016 = df_2016[df_2016["mental_health_benefits"]!="Not eligible for coverage or N/A"]["mental_health_benefits"]
mentalhealthcarecoverage_2016 = mentalhealthcarecoverage_2016.value_counts()/mentalhealthcarecoverage_2016.shape[0]
mentalhealthcarecoverage_2016


mental_health_benefits
Yes           0.499529
Don't know    0.299717
No            0.200754
Name: count, dtype: float64

In [6]:
mentalhealthcarecoverage_2017 = df_2017[df_2017["mental_health_benefits"]!="Not eligible for coverage or N/A"]["mental_health_benefits"]
mentalhealthcarecoverage_2017 = mentalhealthcarecoverage_2017.value_counts()/mentalhealthcarecoverage_2017.shape[0]
mentalhealthcarecoverage_2017

mental_health_benefits
Yes           0.581848
Don't know    0.270665
No            0.147488
Name: count, dtype: float64

In [7]:
mentalhealthcarecoverage_2018 = df_2018[df_2018["mental_health_benefits"]!="Not eligible for coverage or N/A"]["mental_health_benefits"]
mentalhealthcarecoverage_2018 = mentalhealthcarecoverage_2018.value_counts()/mentalhealthcarecoverage_2018.shape[0]
mentalhealthcarecoverage_2018

mental_health_benefits
Yes           0.626471
Don't know    0.250000
No            0.123529
Name: count, dtype: float64

In [8]:
mentalhealthcarecoverage_2019 = df_2019[df_2019["mental_health_benefits"]!="Not eligible for coverage or N/A"]["mental_health_benefits"]
mentalhealthcarecoverage_2019 = mentalhealthcarecoverage_2019.value_counts()/mentalhealthcarecoverage_2019.shape[0]
mentalhealthcarecoverage_2019

mental_health_benefits
Yes           0.573427
Don't know    0.304196
No            0.122378
Name: count, dtype: float64

In [9]:
# whether they are aware of the options for their mental health care available under their employer-provided coverage
AwarenessOfOptions_2014 = df_2014["mental_health_benefits_awareness"].value_counts()/shape14
AwarenessOfOptions_2014

mental_health_benefits_awareness
No          0.406402
Yes         0.338506
Not sure    0.255092
Name: count, dtype: float64

In [10]:
AwarenessOfOptions_2016 = df_2016["mental_health_benefits_awareness"].value_counts()/shape16
AwarenessOfOptions_2016

mental_health_benefits_awareness
No          0.308566
Not sure    0.307692
Yes         0.267483
Name: count, dtype: float64

In [11]:
AwarenessOfOptions_2017 = df_2017["mental_health_benefits_awareness"].value_counts()/shape17
AwarenessOfOptions_2017

mental_health_benefits_awareness
No     0.479005
Yes    0.416796
Name: count, dtype: float64

In [12]:
AwarenessOfOptions_2018 = df_2018["mental_health_benefits_awareness"].value_counts()/shape18
AwarenessOfOptions_2018

mental_health_benefits_awareness
Yes    0.454294
No     0.443213
Name: count, dtype: float64

In [13]:
AwarenessOfOptions_2019 = df_2019["mental_health_benefits_awareness"].value_counts()/shape19
AwarenessOfOptions_2019

mental_health_benefits_awareness
No     0.486842
Yes    0.427632
Name: count, dtype: float64

In [14]:
# whether their employer ever formally discussed mental health
empDisc_2014 = df_2014["employer_mental_health_discussion"].value_counts()/shape14
empDisc_2014

employer_mental_health_discussion
No            0.660524
Yes           0.185257
Don't know    0.154219
Name: count, dtype: float64

In [15]:
empDisc_2016 = df_2016["employer_mental_health_discussion"].value_counts()/shape16
empDisc_2016

employer_mental_health_discussion
No            0.708916
Yes           0.201049
Don't know    0.090035
Name: count, dtype: float64

In [16]:
empDisc_2017 = df_2017["employer_mental_health_discussion"].value_counts()/shape17
empDisc_2017

employer_mental_health_discussion
No            0.636081
Yes           0.251944
Don't know    0.111975
Name: count, dtype: float64

In [17]:
empDisc_2018 = df_2018["employer_mental_health_discussion"].value_counts()/shape18
empDisc_2018

employer_mental_health_discussion
No            0.631579
Yes           0.285319
Don't know    0.083102
Name: count, dtype: float64

In [18]:
empDisc_2019 = df_2019["employer_mental_health_discussion"].value_counts()/shape19
empDisc_2019

employer_mental_health_discussion
No            0.578947
Yes           0.342105
Don't know    0.078947
Name: count, dtype: float64

In [19]:
# whether their employer offer resources to learn more about mental health concerns and options for seeking help
empRes_2014 = df_2014["employer_mental_health_learning_resources"].value_counts()/shape14
empRes_2014 

employer_mental_health_learning_resources
No            0.483996
Don't know    0.310378
Yes           0.205626
Name: count, dtype: float64

In [20]:
empRes_2016 = df_2016["employer_mental_health_learning_resources"].value_counts()/shape16
empRes_2016

employer_mental_health_learning_resources
No            0.462413
Don't know    0.279720
Yes           0.257867
Name: count, dtype: float64

In [21]:
empRes_2017 = df_2017["employer_mental_health_learning_resources"].value_counts()/shape17
empRes_2017

employer_mental_health_learning_resources
No            0.433904
Yes           0.292379
Don't know    0.273717
Name: count, dtype: float64

In [22]:
empRes_2018 = df_2018["employer_mental_health_learning_resources"].value_counts()/shape18
empRes_2018

employer_mental_health_learning_resources
No            0.398892
Yes           0.324100
Don't know    0.277008
Name: count, dtype: float64

In [23]:
empRes_2019 = df_2019["employer_mental_health_learning_resources"].value_counts()/shape19
empRes_2019

employer_mental_health_learning_resources
No            0.427632
Yes           0.335526
Don't know    0.236842
Name: count, dtype: float64

data = [
    {'Year': 2014, 'Response': "Yes", 'mentalhealthcarecoverage': 0.414161, 'AwarenessOfOptions': 0.338506},
    {'Year': 2014, 'Response': "No", 'mentalhealthcarecoverage':  0.246363, 'AwarenessOfOptions': 0.406402},
    {'Year': 2014, 'Response': "Don\'t Know", 'mentalhealthcarecoverage': 0.339476, 'AwarenessOfOptions': 0.255092},
    
    {'Year': 2016, 'Response': "Yes", 'mentalhealthcarecoverage': 0.499529, 'AwarenessOfOptions': 0.267483},
    {'Year': 2016, 'Response': "No", 'mentalhealthcarecoverage': 0.200754, 'AwarenessOfOptions': 0.308566},
    {'Year': 2016, 'Response': "Don\'t Know", 'mentalhealthcarecoverage': 0.299717, 'AwarenessOfOptions': 0.307692},
     
    {'Year': 2017, 'Response': "Yes", 'mentalhealthcarecoverage': 0.581848, 'AwarenessOfOptions': 0.416796},
    {'Year': 2017, 'Response': "No", 'mentalhealthcarecoverage': 0.147488, 'AwarenessOfOptions': 0.479005},
    {'Year': 2017, 'Response': "Don\'t Know", 'mentalhealthcarecoverage': 0.270665, 'AwarenessOfOptions': 0},
     
    {'Year': 2018, 'Response': "Yes", 'mentalhealthcarecoverage': 0.626471, 'AwarenessOfOptions': 0.454294},
    {'Year': 2018, 'Response': "No", 'mentalhealthcarecoverage': 0.123529, 'AwarenessOfOptions': 0.443213},
    {'Year': 2018, 'Response': "Don\'t Know", 'mentalhealthcarecoverage': 0.250000, 'AwarenessOfOptions': 0},
     
    {'Year': 2019, 'Response': "Yes", 'mentalhealthcarecoverage':  0.573427, 'AwarenessOfOptions': 0.427632},
    {'Year': 2019, 'Response': "No", 'mentalhealthcarecoverage': 0.122378, 'AwarenessOfOptions': 0.486842},
    
     {'Year': 2019, 'Response': "Don\'t Know", 'mentalhealthcarecoverage': 0.304196, 'AwarenessOfOptions': 0}]

const csvData = `year,response,mentalhealthcarecoverage,awarenessOfOptions,empDisc,empRes
"2014",Yes,0.414161,0.338506,0.185257,0.205626
"2014",No,0.246363,0.406402,0.660524,0.483996
"2014",Don't Know,0.339476,0.255092,0.154219,0.310378
"2016",Yes,0.499529,0.267483,0.201049, 0.257867
"2016",No,0.200754,0.308566,0.708916, 0.462413
"2016",Don't Know,0.299717,0.307692,0.090035, 0.279720
"2017",Yes,0.581848,0.416796,0.251944, 0.292379
"2017",No,0.147488,0.479005,0.636081, 0.433904
"2017",Don't Know,0.270665,0.000000,0.111975, 0.273717
"2018",Yes,0.626471,0.454294,0.285319,0.324100
"2018",No,0.123529,0.443213,0.631579,0.398892
"2018",Don't Know,0.250000,0.000000,0.083102,0.277008
"2019",Yes,0.573427,0.427632,0.342105, 0.335526
"2019",No,0.122378,0.486842,0.578947,0.427632
"2019",Don't Know,0.304196,0.000000,0.078947,0.236842`;

SIZE OF COMPANY!!!!

In [24]:
size = df[df['number_of_employees'] != 'No Response']
size

year self_employed number_of_employees tech_company  \
0     2014            No             100-500          Yes   
1     2014            No              26-100          Yes   
2     2014            No      More than 1000           No   
3     2014            No              26-100          Yes   
4     2014            No      More than 1000           No   
...    ...           ...                 ...          ...   
3478  2019            No              26-100          Yes   
3479  2019            No                6-25          Yes   
3480  2019            No      More than 1000           No   
3481  2019            No      More than 1000          Yes   
3482  2019            No      More than 1000          Yes   

     mental_health_benefits mental_health_benefits_awareness  \
0                       Yes                              Yes   
1                       Yes                              Yes   
2                Don't know                               No   
3                       Yes                         Not sure   
4                       Yes                              Yes   
...                     ...                              ...   
3478                    Yes                               No   
3479                     No                               No   
3480             Don't know                               No   
3481                     No                               No   
3482                    Yes                              Yes   

     employer_mental_health_discussion  \
0                                   No   
1                                   No   
2                                   No   
3                           Don't know   
4                                   No   
...                                ...   
3478                                No   
3479                                No   
3480                                No   
3481                                No   
3482                                No   

     employer_mental_health_learning_resources  \
0                                           No   
1                                           No   
2                                   Don't know   
3                                          Yes   
4                                   Don't know   
...                                        ...   
3478                                        No   
3479                                        No   
3480                                Don't know   
3481                                        No   
3482                                       Yes   

     mental_health_treatment_anonymity mental_health_leave_accessibility  \
0                                   No                Somewhat difficult   
1                           Don't know                        Don't know   
2                           Don't know                        Don't know   
3                                  Yes                        Don't know   
4                                   No                         Very easy   
...                                ...                               ...   
3478                               Yes                         Very easy   
3479                        Don't know                         Very easy   
3480                        Don't know                Somewhat difficult   
3481                               Yes                         Difficult   
3482                               Yes                Somewhat difficult   

     mental_health_discussion_comfort_coworkers  \
0                                  Some of them   
1                                  Some of them   
2                                            No   
3                                           Yes   
4                                  Some of them   
...                                         ...   
3478                                        Yes   
3479                                         No   
3480                      

In [25]:
size.groupby(["number_of_employees","employer_mental_health_discussion"]).count()

year  self_employed  \
number_of_employees employer_mental_health_discussion                        
1-5                 Don't know                           11             11   
                    No                                  122            122   
                    Yes                                  17             17   
100-500             Don't know                           99             99   
                    No                                  488            488   
                    Yes                                 181            181   
26-100              Don't know                           61             61   
                    No                                  597            597   
                    Yes                                 131            131   
500-1000            Don't know                           30             30   
                    No                                  135            135   
                    Yes                                  78             78   
6-25                Don't know                           37             37   
                    No                                  546            546   
                    Yes                                  58             58   
More than 1000      Don't know                          150            150   
                    No                                  416            416   
                    Yes                                 325            325   

                                                       tech_company  \
number_of_employees employer_mental_health_discussion                 
1-5                 Don't know                                   11   
                    No                                          122   
                    Yes                                          17   
100-500             Don't know                                   99   
                    No                                          488   
                    Yes                                         181   
26-100              Don't know                                   61   
                    No                                          597   
                    Yes                                         131   
500-1000            Don't know                                   30   
                    No                                          135   
                    Yes                                          78   
6-25                Don't know                                   37   
                    No                                          546   
                    Yes                                          58   
More than 1000      Don't know                                  150   
                    No                                          416   
                    Yes                                         325   

                                                       mental_health_benefits  \
number_of_employees employer_mental_health_discussion                           
1-5                 Don't know                                             11   
                    No                                                    122   
                    Yes                                                    17   
100-500             Don't know                                             99   
                    No                                                    488   
                    Yes                                                   181   
26-100              Don't know                                             61   
                    No                                                    597   
                    Yes                                                   131   
500-1000            Don't know                                             30   
                    No                                                    135   
                    

In [26]:
size["number_of_employees"].value_counts()

number_of_employees
More than 1000    891
26-100            789
100-500           768
6-25              641
500-1000          243
1-5               150
Name: count, dtype: int64

In [27]:
16/150

0.10666666666666667

In [28]:
54/641

0.0842433697347894

In [29]:
119/789

0.15082382762991128

In [30]:
213/768

0.27734375

In [31]:
85/243

0.3497942386831276

In [32]:
427/891

0.4792368125701459

const csvData = `year,response,mentalhealthcarecoverage,awarenessOfOptions,empDisc,empRes
"1-5",Yes,0.14,0.22666666666666666,0.11333333333333333,0.10666666666666667
"6-25",Yes,0.2792511700468019,0.21996879875195008,0.0904836193447738, 0.0842433697347894
"26-100",Yes,0.42712294043092525,0.30925221799746516,0.16603295310519645, 0.15082382762991128
"100-500",Yes,0.5442708333333334,0.3854166666666667,0.23567708333333334,0.27734375
"500-1000",Yes,0.5843621399176955,0.41975308641975306,0.32098765432098764, 0.3497942386831276
"More than 1000",Yes,0.6689113355780022,0.4489337822671156,0.36475869809203143, 0.4792368125701459

PIE CHART

In [35]:
df

year self_employed number_of_employees tech_company  \
0     2014            No             100-500          Yes   
1     2014            No              26-100          Yes   
2     2014            No      More than 1000           No   
3     2014            No              26-100          Yes   
4     2014            No      More than 1000           No   
...    ...           ...                 ...          ...   
3478  2019            No              26-100          Yes   
3479  2019            No                6-25          Yes   
3480  2019            No      More than 1000           No   
3481  2019            No      More than 1000          Yes   
3482  2019            No      More than 1000          Yes   

     mental_health_benefits mental_health_benefits_awareness  \
0                       Yes                              Yes   
1                       Yes                              Yes   
2                Don't know                               No   
3                       Yes                         Not sure   
4                       Yes                              Yes   
...                     ...                              ...   
3478                    Yes                               No   
3479                     No                               No   
3480             Don't know                               No   
3481                     No                               No   
3482                    Yes                              Yes   

     employer_mental_health_discussion  \
0                                   No   
1                                   No   
2                                   No   
3                           Don't know   
4                                   No   
...                                ...   
3478                                No   
3479                                No   
3480                                No   
3481                                No   
3482                                No   

     employer_mental_health_learning_resources  \
0                                           No   
1                                           No   
2                                   Don't know   
3                                          Yes   
4                                   Don't know   
...                                        ...   
3478                                        No   
3479                                        No   
3480                                Don't know   
3481                                        No   
3482                                       Yes   

     mental_health_treatment_anonymity mental_health_leave_accessibility  \
0                                   No                Somewhat difficult   
1                           Don't know                        Don't know   
2                           Don't know                        Don't know   
3                                  Yes                        Don't know   
4                                   No                         Very easy   
...                                ...                               ...   
3478                               Yes                         Very easy   
3479                        Don't know                         Very easy   
3480                        Don't know                Somewhat difficult   
3481                               Yes                         Difficult   
3482                               Yes                Somewhat difficult   

     mental_health_discussion_comfort_coworkers  \
0                                  Some of them   
1                                  Some of them   
2                                            No   
3                                           Yes   
4                                  Some of them   
...                                         ...   
3478                                        Yes   
3479                                         No   
3480                      

In [44]:
df.groupby(["tech_company", "employer_mental_health_learning_resources"]).count()

year  self_employed  \
tech_company employer_mental_health_learning_resources                        
No           Don't know                                  206            206   
             No                                          306            306   
             Yes                                         284            284   
Yes          Don't know                                  782            782   
             No                                         1275           1275   
             Yes                                         630            630   

                                                        number_of_employees  \
tech_company employer_mental_health_learning_resources                        
No           Don't know                                                 206   
             No                                                         306   
             Yes                                                        284   
Yes          Don't know                                                 782   
             No                                                        1275   
             Yes                                                        630   

                                                        mental_health_benefits  \
tech_company employer_mental_health_learning_resources                           
No           Don't know                                                    206   
             No                                                            306   
             Yes                                                           284   
Yes          Don't know                                                    782   
             No                                                           1275   
             Yes                                                           630   

                                                        mental_health_benefits_awareness  \
tech_company employer_mental_health_learning_resources                                     
No           Don't know                                                              198   
             No                                                                      279   
             Yes                                                                     268   
Yes          Don't know                                                              746   
             No                                                                     1117   
             Yes                                                                     612   

                                                        employer_mental_health_discussion  \
tech_company employer_mental_health_learning_resources                                      
No           Don't know                                                               206   
             No                                                                       306   
             Yes                                                                      284   
Yes          Don't know                                                               782   
             No                                                                      1275   
             Yes                                                                      630   

                                                        mental_health_treatment_anonymity  \
tech_company employer_mental_health_learning_resources                                      
No           Don't know                                                               206   
             No                                                                       306   
             Yes                                                                      284   
Yes          Don't know                                                               782   
             No                                                                      1275   


Out of the percentageof people who have said yes

const csvData3 = `istechcomp,mentalHealthCoverage,awarenessOfOptions,employeerDiscussion,employeerResources
"No",410,298,238,284
"Yes",1283,919,238,630;